server.js

```diff
@@ -6801,6 +6801,38 @@ cache((data, match, sendBadge, request) => {
   });
 }));
 
+// nsp for npm packages
+camp.route(/^\/nsp\/npm\/([^\/]+)\.(svg|png|gif|jpg|json)$/, cache((data, match, sendBadge, request) => {
+  const package = match[1];
+  const format = match[2];
+  let badgeData = getBadgeData('nsp', data);
+
+  request(`https://api.nodesecurity.io/check/${package}`, (error, response, body) => {
+    let jsonResults;
+
+    if (body !== null) {
+      try {
+        jsonResults = JSON.parse(body);
+      } catch (e) {}
+    }
+
+    if (response !== null && response.statusCode === 404) {
+      badgeData.text[1] = 'invalid';
+    } else if (error !== null || jsonResults === undefined) {
+      badgeData.text[1] = 'error';
```

Comment:
Though **error** might be more descriptive, for consistency, this one should be **invalid**. The one before should be something like **package not found**.

By the way, does NSP support scoped packages like [@cycle/core](https://www.npmjs.com/package/@cycle/core)? If so could you add a test to make sure it works for that too? It might require different quoting or a different regex.

Reply:
It looks like NSP [supports scoped packages](https://github.com/nodesecurity/nsp/issues/192#issuecomment-332293289). I'll update the implementation to support them.

Reply:
I'll also add support for specific package versions - perhaps `/nsp/npm/l/:package.:format` for the latest version and `/nsp/npm/v/:package.:format` for a specific version?

Reply:
To be consistent with the other badges, `/nsp/npm/:package.:format` for latest and `/nsp/npm/:package/:version.:format` for specific.

`/l/` usually means license and `/v/` means version. The equivalent here would be something like `/nsp/count/npm/`, if you thought there might be badges for other kinds of nsp data in the future.